stdlib/uri/0/mailto.rbs

```diff
@@ -4,5 +4,89 @@ module URI
   # RFC6068, the mailto URL scheme.
   #
   class MailTo < Generic
+    EMAIL_REGEXP: Regexp
+
+    # <!--
+    #   rdoc-file=lib/uri/mailto.rb
+    #   - build(args)
+    # -->
+    # ## Description
+    #
+    # Creates a new URI::MailTo object from components, with syntax checking.
+    #
+    # Components can be provided as an Array or Hash. If an Array is used, the
+    # components must be supplied as `[to, headers]`.
+    #
+    # If a Hash is used, the keys are the component names preceded by colons.
+    #
+    # The headers can be supplied as a pre-encoded string, such as
+    # `"subject=subscribe&cc=address"`, or as an Array of Arrays like `[['subject',
+    # 'subscribe'], ['cc', 'address']]`.
+    #
+    # Examples:
+    #
+    #     require 'uri'
+    #
+    #     m1 = URI::MailTo.build(['joe@example.com', 'subject=Ruby'])
+    #     m1.to_s  # => "mailto:joe@example.com?subject=Ruby"
+    #
+    #     m2 = URI::MailTo.build(['john@example.com', [['Subject', 'Ruby'], ['Cc', 'jack@example.com']]])
+    #     m2.to_s  # => "mailto:john@example.com?Subject=Ruby&Cc=jack@example.com"
+    #
+    #     m3 = URI::MailTo.build({:to => 'listman@example.com', :headers => [['subject', 'subscribe']]})
+    #     m3.to_s  # => "mailto:listman@example.com?subject=subscribe"
+    #
+    def self.build: (Array[String]) -> instance
+                  | ([String, Array[Array[String]]]) -> instance
+                  | (Hash[Symbol, String | Array[Array[String]]]) -> instance
+
+    # <!-- rdoc-file=lib/uri/mailto.rb -->
+    # E-mail headers set by the URL, as an Array of Arrays.
+    #
+    def headers: () -> Array[[String, String]]
```

Comment:
I found invalid pattern by raap.

``````
$ bundle exec raap -r uri --library uri URI::MailTo#headers
[RaaP] INFO: # URI::MailTo
[RaaP] INFO: ## def headers: () -> ::Array[[ ::String, ::String ]]
EEF

() -> Array[[String, String]]

[RaaP] INFO: success: 0, skip: 0, exception: 2, time: 15ms
# 1) Failure:

def headers: () -> ::Array[[ ::String, ::String ]]
  in /Users/ksss/src/github.com/ksss/rbs/stdlib/uri/0/mailto.rbs:46:17...46:46

## Reason

Failed in case of `#<URI::MailTo>.headers() -> [["ez"]][Array[Array[String]]]`

### Repro

```rb
uri_mailto = URI::MailTo.new(nil, nil, 'lm', -4, nil, nil, nil, 'ez', nil)
uri_mailto.headers()
```
``````

Reply:
Looks like the `query` (`?ez`) cannot be constructed with `URI::MailTo.build`. 

```
irb(main):020> URI::MailTo.build(["hoge@huga", "foo"]).headers
/Users/soutaro/.rbenv/versions/3.3.2/lib/ruby/3.3.0/uri/mailto.rb:212:in `check_headers': bad component(expected opaque component): foo (URI::InvalidComponentError)
```

I think we can have `Array[[String, String]]` here, even raap detects an error.